wurst/event/DamageModify.wurst

```diff
@@ -0,0 +1,327 @@
+/**
+	DamageModify
+	by Trokkin
+
+	API:
+
+	// Pretty much '(Damage d) -> ...'
+	interface ForDamageCallback
+		run(Damage d)
+
+	// these are called before the damage is dealt
+	// can be called multiple amount of times or not called at all
+	addDamageModifier(ForDamageCallback cb)
+	removeDamageModifier(ForDamageCallback cb)
+	
+	// these are called after the damage is dealt
+	// are called strictly once per damage instance
+	addDamageListener(ForDamageCallback cb)
+	removeDamageListener(ForDamageCallback cb)
+
+	// a short for 'new Damage(this, target, amount, isSpell, id)..modify()..deal()'
+	unit.damage(unit target, real amount, OPTIONAL bool isSpell = true, int id = 0)
```

Comment:
I'm a bit confused by `OPTIONAL` here, in particular because you haven't labeled `id` as optional but it has a defaulted value.

Suggest just removing `OPTIONAL` qualifier.

What is `id` for?

What effect does `isSpell` have?

Reply:
What does `modify` method do? Does it need to be introduced before this docu?